### PR TITLE
Ignore Backstage catalog-info.yaml files

### PR DESCRIPTION
**Pull request by bautrey37**
These files store the metadata of the repository which are then
harvested and visualized in Backstage. These files do not affect
deployment of software and thus do not need to trigger a deploy.

[backstage-catelog-overview](https://backstage.io/docs/features/software-catalog/software-catalog-overview)

### CODE REVIEW

File: .deployignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 README.adoc
 LICENSE.md
+catelog-info.yaml
```
Comment (sturmer) [inline .deployignore:3]: Typo? (catelog -> catalog)

Comment (bautrey37) [reply .deployignore:3]: Yes! Thank you, very important!

**Review by sturmer — APPROVED** [review]

**Comment by bautrey37** [comment]
`review/squash` is not being reported. Not sure why. I'll push force blank changes to try to get github to report a status on this.

**Comment by bautrey37** [comment]
I misunderstood how this deployignore feature worked. I thought this was a global deployignore, but it is not. [See](https://github.com/salemove/engagement-router/pull/674#issuecomment-1085785076). This is affects the current repository. This change can stay as eventually this repo will have a `catalog-info.yaml` file.

**Pull request merged**